test/rhsmlib/facts/test_dmifacts.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2022 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+
+from rhsmlib.facts.dmidecodeparser import DmidecodeParser
+from rhsmlib.facts.dmiinfo import DmidecodeFactCollector
+
+import contextlib
+import os
+import unittest
+
+
+FAKE_PART_NUMBER = "AAAAAAAAAAAAA-ZZ"
+FAKE_PRODUCT_NAME = "PPPPPPPPPP"
+FAKE_SERIAL_NUMBER = "SSSSSSSS"
+FAKE_UUID = "11111111-2222-3333-4444-555555555555"
+
+
+class DmidecodeTestDataMixin:
+    """
+    Simple mixin for testing dmidecode output files.
+    """
+
+    DATA_SUFFIX = ".txt"
+
+    @property
+    def datadir(self):
+        curdir = os.path.dirname(os.path.realpath(__file__))
+        dmidecodedatadir = os.path.join(curdir, "dmidecodedata")
+        return dmidecodedatadir
+
+    def load_data(self, name):
+        if not name.endswith(self.DATA_SUFFIX):
+            name += self.DATA_SUFFIX
```

Comment:
Is it necessary to add suffix to name?

Reply:
my idea was to leave that, so it is possible to do in the future e.g.:
```python
    parser_of_file = self.load_data("x86_64-baremetal-1")
    # do some specific testing with the data of that specific file
```
and not hardcoding the extension was simply a way to have one data less hardcoded in the tests.

(so far I didn't add any test like the one above)